### ISSUE TITLE
[Accessibility] Add a "Skip To Content" button to `SubdomainNavBar`

### ISSUE THREAD

**Issue by simurai** [primer/brand#404]
**Note**: This is not really urgent, but still thought it might be a "nice to have" at some point, since I assume we will run into it for every site/page we'll build.

---

This came up in https://github.com/github/accessibility-audits/issues/5220, where we got asked to add a "Skip To Content" button that is shown when a user uses the `tab` key to navigate and wants to skip having to go through all the options of the nav (`SubdomainNavBar`).

![Button in the top left corner with a Skip To Content label](https://github.com/github/githubuniverse.com/assets/378023/3303e661-3547-46a0-9edf-3f920398e041)

Here is the [implementation for Universe](https://github.com/github/githubuniverse.com/pull/1805). The button could be the first item in the `SubdomainNavBar`, but not sure about adding an `id="start-of-content"` that the button links to. For Universe we added it on the following [`<main>` element](https://github.com/github/githubuniverse.com/pull/1805/files#diff-273e11bd53a1ddfe40b66fd62d9b9b70337ab54bd39ab1718f43e8999198aee1R32). But ideally that should happen automatically without the need to remember it. Hmm..  Maybe since it's essentially just a "skip the nav", we might can also add an empty `<div id="start-of-content"` at the very end of `SubdomainNavBar`? 🤔

**Comment by stamat** [comment]
@simurai I was gonna say that we should have a main element too, maybe we should introduce it to dotcom marketing pages. Otherwise we could query the first paragraph maybe? Or since we now use React maybe find an element that we can latch onto. Or maybe we detect the hero and then when you click we scroll down under the hero. I'll give it a thought, and I'll report back!
Maybe there is some noninvasive way to do this. But the logic should be run on click event.

**Comment by stamat** [comment]
@simurai after looking at it... And realizing that skiping to content actually means skipping the header menu lol, I was about to suggest the same:

> Maybe since it's essentially just a "skip the nav", we might can also add an empty <div id="start-of-content" at the very end of SubdomainNavBar?

And the code already does this if the `main` with the ID is not available! Man I gotta update myself on React lol...

**Comment by simurai** [comment]
> @stamat And realizing that skiping to content actually means skipping the header menu

Yeah, I think it's mostly for users that already know the nav and don't want to tab through it every time they land on a page. So instead of having to `tab tab tab tab...` they can just do `tab enter` and skip the nav.

> And the code already does this if the `main` with the ID is not available!

Ohh.. You mean the `<SubdomainNavBar>` component already skips to the next `main` element? Yeah, that might be even better as a default. Because in the current GitHub Universe implementation, when a user clicks on "Skip to content", it links to https://githubuniverse.com/#start-of-content. And then the user might copy and paste that URL when sharing, which might not be ideal. So skipping to the content without using `#start-of-content` would be nice.. if that's possible. 🤔

**Comment by stamat** [comment]
@simurai hey man ✨  I've made the final changes that Rezah requested from the previous developer. Just a to see why AXE CI is failing for two test stories and we are good to go!

To explain a bit better, anchoring is used either way. And either way you will need an ID, this is because we want to leave it to the browser to handle the scroll to and leverage `scroll-behavior: smooth` property (if the user approves of web movement).

- So if there is no `main` element a `div id="start-of-content" will be created. 
- If there is a `main` element, but without the id, it will add the `start-of-content` id to the element.
- If there is a `main` element with the id, for instance `my-id`, the anchor will become `#my-id`.

I can of course make a function that will target the element offset and set scrollTop to it. And then see if the web movement is preferred and set scroll behavior accordingly.  Though I think it's an overkill?

**Comment by simurai** [comment]
> * So if there is no `main` element a `div id="start-of-content" will be created.
> * If there is a `main` element, but without the id, it will add the `start-of-content` id to the element.
> * If there is a `main` element with the id, for instance `my-id`, the anchor will become `#my-id`.

Perfect! 🙌

> @stamat I can of course make a function that will target the element offset and set scrollTop to it. And then see if the web movement is preferred and set scroll behavior accordingly. Though I think it's an overkill?

Yeah.. probably overkill. 👍  Using the built in "scroll to anchor" sounds good to me and the risk of "copy and paste the deep link" feels minor.
